### PR TITLE
Adding ability to edit an annoucement.

### PR DESCRIPTION
**Pull request by LeCoffee**
Added an example.

### CODE REVIEW

**Comment by demipixel** [comment]
Documentation? :3

**Comment by LeCoffee** [comment]
Will edit the wiki when its actually implemented. For now their is an example file but its just

```javascript 
group.editAnnouncement(aid, header, content, function(error) {
   if(!error) {
     console.log("Annoucement edited!");
   } else {
     console.log("Unable to edit annoucement! %j", error);
     process.exit(1);
   }
 });
```

where AID is the announcement ID (The long number when looking at a detailed announcement)
Header is just a string for the title
Content is another string but for the content.

You must get the group object from 
```
community.getSteamGroup('id', function(err, group) {});
```

**Comment by andrewda** [comment]
Why does this change so much spacing in CSteamGroup.js?

**Comment by LeCoffee** [comment]
Honestly, I have no idea. Didn't notice it until it showed up on github. I'm trying out the new Atom.io editor I presume it automatically got rid of the empty lines that was just spaces and replaced them with nothing, probably for making client side loading slightly faster for web development. I can make a commit to add white spaces for no reason if you like :)

**Comment by andrewda** [comment]
Just my OCD kicking in, I think it would be best to keep the code clean and regular throughout. I'll also add that the spacing in [edit-group-announcements.js](https://github.com/LeCoffee/node-steamcommunity/blob/master/examples/edit-group-announcement.js#L17-L79) gets a little wacky throughout. Tabs should be used for indentation.

**Comment by LeCoffee** [comment]
Yeah, just checked is atom.io also doesn't show the difference between tabs and spaces apparently hence I didn't see the tabs/space difference, I'm also OCD about this and just assumed atom would keep everything consistent like sublime (at least it shows that). I use tabs for most things however when I press tab in atom it adds two spaces while the original example had spaces. Not really an issue I'll open sublime and fix it :)

**Comment by andrewda** [comment]
Great, should add a nice amount of readability!

:+1: for this feature, btw.

**Comment by LeCoffee** [comment]
https://github.com/LeCoffee/node-steamcommunity/blob/7627e6a3a0d2c12ee9a3bc50a62359ca8e62bcdc/components/groups.js

Just noticed groups.js doesn't seem to follow the same pattern. I'll just leave it how it is for now.

File: classes/CSteamGroup.js [hunk 1]
```diff
@@ -64,7 +64,7 @@ CSteamGroup.prototype.getMembers = function(addresses, callback) {
 		callback = addresses;
 		addresses = null;
 	}
-
+
```
Comment (scholtzm) [inline classes/CSteamGroup.js]: You missed this one. :grinning:

**Comment by LeCoffee** [comment]
I hope your kidding, I've spend 5 minutes looking at the same line with it in :laughing:

**Comment by andrewda** [comment]
CSteamGroup.js is styled differently than the rest for some reason. There's no need for the extra tab. (Styling on point)

**Comment by LeCoffee** [comment]
Just added a small method to pull in all the announcements found in the RSS feed. 

Things to note:
* You can specify an optional time as the first argument
* The "author" is a persona name NOT a steamid

Possible things to add:
* group.getAnnouncement(aid, callback)
This could get more information about the announcement such as the authors SteamID and comments. 
Another thing it could return a CSteamAnnouncement object (Like the CSteamGroup) where other methods could be built in such as up-voting, down-voting commenting etc. I'd like to implement these but I don't know if I have the time right now and would like feedback on creating a CSteamAnnouncement.

Just mentioning what I mentioned in comments and #69

File: components/groups.js [hunk 2]
```diff
@@ -125,6 +125,62 @@ SteamCommunity.prototype.leaveGroup = function(gid, callback) {
 	});
 };
 
+SteamCommunity.prototype.getAllAnnouncements = function(gid, time, callback) {
```
Comment (DoctorMcKay) [inline components/groups.js]: This needs to have "group" in the method name to make it apparent where it's getting announcements from. For example, `getAllGroupAnnouncements`.

File: components/groups.js [hunk 3]
```diff
@@ -125,6 +125,62 @@ SteamCommunity.prototype.leaveGroup = function(gid, callback) {
...
+		callback = time;
+		time = new Date(0); // The beginnig of time...
+	}
+
+	var self = this;
+	this.request({ 
+		"uri": "https://steamcommunity.com/gid/" + gid.getSteamID64() + "/rss/" 
+	}, function(err, response, body) {
+		if(!callback) {
+			return;
```
Comment (DoctorMcKay) [inline components/groups.js]: This doesn't really hurt anything, but there's no need to make the callback of a get request optional.

File: components/groups.js [hunk 4]
```diff
@@ -125,6 +125,62 @@ SteamCommunity.prototype.leaveGroup = function(gid, callback) {
...
+		}
+
+		if(err || response.statusCode >= 400) {
```
Comment (DoctorMcKay) [inline components/groups.js]: Should use `self._checkHttpError` instead.

File: components/groups.js [hunk 5]
```diff
@@ -125,6 +125,62 @@ SteamCommunity.prototype.leaveGroup = function(gid, callback) {
...
+
+			var announcements = results.rss.channel[0].item.map(function(announcement) {
+				var splitLink = announcement.link[0].split('/');
+				return {
+					headline: announcement.title[0],
+					content:  announcement.description[0],
+					date:     new Date(announcement.pubDate[0]),
+					author:   announcement.author[0], // Unfortunately, the RSS feed likes to give us personanames not steamid's
+					aid:      splitLink[splitLink.length - 1] // The ID after the last /
+					// Note this is marked as guid (gid?) in the rss feed but can also be obtained from link
```
Comment (DoctorMcKay) [inline components/groups.js]: Long numbers in Steam that aren't SteamIDs are nearly always GIDs (GlobalIDs). They're global to the Steam system and contain a boxID (the ID of the server where they were generated), startup time of the server that generated it, and a sequential counter.

**Comment by LeCoffee** [comment]
The GID meaning GlobalID makes much more sense, I just assumed it meant group ID since i've only really been in the groups side of this.

As for the self._checkHttpError(err, response, callback) it wasn't in the postGroupAnnoucement so I didn't add it to the others. I've gone ahead and replaced the original check of > 400 etc with this in postAnnouncement editAnnouncement and getAllAnnouncements. I did see it in schedule event however didn't want to make this PR do more than what was stated.

I've also gone ahead and added Group into the getAllGroupAnnouncements and removed the optional callback since they're calling the method they clearly want data back.

**Comment by DoctorMcKay** [comment]
Thanks! Out in v3.18.0.

**Pull request merged**